### PR TITLE
file-editor question type implementation

### PR DESCRIPTION
**Pull request by amurph24**
Updates to problem_bank_scripts.py:
- will add autograde test directory file where relevant
- add def_process_code input function for code input questions

### CODE REVIEW

File: src/problem_bank_scripts/problem_bank_scripts.py [hunk 1]
```diff
@@ -690,6 +690,27 @@ def process_file_upload(part_name, parsed_question, data_dict):
     return replace_tags(html)
 
 
+def process_code_input(part_name, parsed_question, data_dict):
```
Comment (firasm) [inline src/problem_bank_scripts/problem_bank_scripts.py:693]: Where possible, I try to use the exact same name as the element, so suggest changing this to `process_file_editor()`

File: src/problem_bank_scripts/problem_bank_scripts.py [hunk 2]
```diff
@@ -1066,6 +1089,13 @@ def process_question_pl(source_filepath, output_path=None):
         pl_path.mkdir(parents=True, exist_ok=True)
         [copy2(pathlib.Path(source_filepath).parent / fl, pl_path / fl) for fl in files_to_copy]
 
+    # Move autograde py test files
+    files_to_copy = parsed_q["header"].get("autogradeTestFiles")
+    if files_to_copy:
+        pl_path = output_path / "tests"
```
Comment (firasm) [inline src/problem_bank_scripts/problem_bank_scripts.py:1095]: It's important to troubleshoot and test this output_path business...it's very finicky.

**Review by firasm — CHANGES_REQUESTED** [review]
- [x] You've tested this change and it works as expected right ? It creates a fully functional auto graded question?

- [x] Can you also add an example in the templates directory in this repo that shows it works?

**Comment by amurph24** [comment]
There changes are to:
- `write_info_json`: previously, only the mandatory fields were written to the info.json file when the .md question was converted. The optional fields listed from lines 418-430 have been implemented to correctly concatenate using variables gathered from the md header.
- `process_question_pl`: files that are included in a dir named 'tests' that is a sibling to the .md question file will now be moved to a tests dir as the question is converted to the PL format. This is done in the same way image assets are extracted, and the file names must be listed under `autogradeTestFiles`.
Newly implemented functions:
- `process_file_editor`: This is a slightly tweaked `process_longtext` function that uses the file-editor html element instead of the rich-text-editor html element, otherwise it is identical to `process_longtext`.

A template can be found in the [instructor_datascience_bank](https://github.com/open-resources/instructor_datascience_bank/tree/add_file_editor_template_question/templates/source/q13_file-editor-input) repository. After it is processed using `checkq.py` the question functions correctly in the DATA301 PL course, the exact question is found [here](https://ca.prairielearn.com/pl/course/54/question/266080/preview).

**Comment by amurph24** [comment]
I am having trouble determining what I should be inserting in the expected outputs for [instructors](https://github.com/open-resources/problem_bank_scripts/tree/main/tests/test_question_templates/question_expected_outputs/instructor) and the [public](https://github.com/open-resources/problem_bank_scripts/tree/main/tests/test_question_templates/question_expected_outputs/public) to add them to the repository, Could you lend me some assistance? as stated above the question generated works correctly within the DATA301 PL environment.

**Comment by firasm** [comment]
The `problem_bank_scripts` exports three types of files at the moment:

- "pl" (this is what you have currently, and is the standard output of `pbs.process_question_pl(source_filepath)`)
- "instructor" (this export type contains the solution and is the output of `pbs.process_question_md(source_filepath, instructor=True)`)
- "public" (this export type contains just the question and possible answers and is the output of `pbs.process_question_md(source_filepath, instructor=False)`).

Here's an example of this script being used : https://github.com/open-resources/instructor_physics_bank/blob/7a211fff1863617fe9e8af0f0a7c816928514a2e/scripts/process.py#L43-L50

**Comment by firasm** [comment]
Okay I think I am going to merge this in for now, and the tests can go into #37 that @Bluesy1 is working on.

**Comment by Bluesy1** [comment]
> Okay I think I am going to merge this in for now, and the tests can go into #37 that @Bluesy1 is working on.

I've updated the tests I'm working to pass with these new problem types in 9ce4c34b071d18fdb86bbc426b07a2c07ef5aa73 - subdirectories caused a bit of an issue initially due to the PrairieLearn test not expecting subdirectories in generated outputs but that's fixed.

Are there any extra test problems we want to add to the set of test problems add based off this PR?

**Comment by firasm** [comment]
I think the templates should cover all of the possibilities, as we add more problem types, the templates will need to be updated.

**Pull request merged**